File: api/app/moments.py
from datetime import UTC, datetime

from app.database import supabase


def register_moment(
    entity_type: str,
    entity_id: int,
    event_type: str,
    value: str | None = None,
    moment_note: str | None = None,
):
    return (
        supabase
        .table("moments")
        .insert({
            "entity_type": entity_type,
            "entity_id": entity_id,
            "event_type": event_type,
            "value": value,
            "moment_note": moment_note,
            "time": datetime.now(UTC).isoformat(),
        })
        .execute()
    )


def enrich_task(task: dict) -> dict:
    previous = (
        supabase
        .table("tasks_sequence")
        .select("task_previous, tasks!task_previous(id, name)")
        .eq("task_next", task["id"])
        .execute()
        .data
    )
    next_ = (
        supabase
        .table("tasks_sequence")
        .select("task_next, tasks!task_next(id, name)")
        .eq("task_previous", task["id"])
        .execute()
        .data
    )
    task["previous_task"] = previous[0]["tasks"] if previous else None
    task["next_task"] = next_[0]["tasks"] if next_ else None
    return task

File: src/lib/api/moments.js
import { supabase } from '../supabaseClient';

function unwrap({ data, error }) {
  if (error) throw error;
  return data;
}

export const momentsApi = {
  listForEntity: (entityType, entityId) =>
    supabase
      .from('moments')
      .select('*')
      .eq('entity_type', entityType)
      .eq('entity_id', entityId)
      .order('time', { ascending: false })
      .then(unwrap),

  addNote: (entityType, entityId, note) =>
    supabase
      .from('moments')
      .insert({
        entity_type: entityType,
        entity_id: entityId,
        moment_type: 'note',
        moment_note: note,
      })
      .select()
      .single()
      .then(unwrap),

  remove: (id) => supabase.from('moments').delete().eq('id', id).then(unwrap),
};
